contrib/filter-extensions/rand_cpp/rand.cpp

```diff
@@ -0,0 +1,105 @@
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+#include <time.h>
+
+extern "C" {
+	#include "../../../list-objects-filter-extensions.h"
+	#include "adapter_functions.h"
+}
+
+struct rand_context {
+	int percentageMatch = 0;
+	int matchCount = 0;
+	int blobCount = 0;
+	int treeCount = 0;
+	struct timespec started_at;
+};
+
+static int rand_init(
+	const struct repository *r,
+	const char *filter_arg,
+	void **context)
+{
+	struct rand_context *ctx = new rand_context();
+
+	ctx->percentageMatch = atoi(filter_arg);
+	if (ctx->percentageMatch > 100 || ctx->percentageMatch < 0) {
+		std::cerr << "filter-rand-cpp: warning: invalid match %: " << filter_arg << "\n";
+		ctx->percentageMatch = 1;  // default 1%
+	}
+	std::cerr << "filter-rand-cpp: matching " << ctx->percentageMatch << "%\n";
+	clock_gettime(CLOCK_MONOTONIC, &ctx->started_at);
+	(*context) = ctx;
+
+	return 0;
+}
+
+enum list_objects_filter_result rand_filter_object(
```

Comment:
```suggestion
static enum list_objects_filter_result rand_filter_object(
```

Otherwise it ends up in the global symbol table at link time (ie: will clash with other names)

A namespace (anonymous or other) around everything except `filter_extension_rand_cpp` works too.

Reply:
Added anonymous namespace around everything

Reply:
should un-static rand_init then